api/grpcserver/http_server.go

```diff
@@ -90,19 +89,21 @@ func (s *JSONHTTPServer) StartService(
 			err = pb.RegisterTransactionServiceHandlerServer(ctx, mux, typed)
 		case *DebugService:
 			err = pb.RegisterDebugServiceHandlerServer(ctx, mux, typed)
+		default:
+			s.logger.Error("unsupported service", zap.Stringer("service", svc))
```

Comment:
This could also be a `s.logger.Panic` to catch it early and not let is slip as this indicates a logic error.

Reply:
It was a mistake in my opinion. Hitting `default` case is clearly a programming mistake and a bug to fix. The earlier it crashes, the less time one will spend debugging.

Reply:
Since the default configuration would lead to a panic on startup if the `JSONListener` is set I updated the `Service` interface to include a register method for the `JSONServer` and updated all existing services to implement this interface.

Now this error cannot occur any more and the `switch` doesn't need to be updated every time a new service is added (just the service needs to implement the interface).